src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -99,6 +150,124 @@ public void OnDiagnosticsUnhandledException(HttpContext httpContext, Exception e
+                }
+            }
+            else
+            {
+                // The request doesn't have the new "Request-Id" header but the standard headers,
+                // we create a new activity and initialize it by the standard headers.
+                StringValues xmsRequestRootId;
+                if (httpContext.Request.Headers.TryGetValue(RequestResponseHeaders.StandardRootIdHeader, out xmsRequestRootId))
+                {
+                    var activity = new Activity(ActivityCreatedByHostingDiagnosticListener);
+                    activity.SetParentId(xmsRequestRootId);
+                    activity.Start();
+                    httpContext.Features.Set(activity);
+                }
+
+                StringValues xmsRequestId;
+                if (httpContext.Request.Headers.TryGetValue(RequestResponseHeaders.StandardParentIdHeader, out xmsRequestId))
+                {
+                    requestTelemetry.Context.Operation.ParentId = xmsRequestId;
+                }
+            }
+
+            requestTelemetry.Id = Activity.Current?.Id;
+            requestTelemetry.Context.Operation.Id = Activity.Current?.RootId;
+
+            this.client.Initialize(requestTelemetry);
+            requestTelemetry.Start();
+            httpContext.Features.Set(requestTelemetry);
+
+            return requestTelemetry;
+        }
+
+        private RequestTelemetry InitializeRequestTelemetryOnBeginRequest(HttpContext httpContext, long timestamp)
+        {
+            var requestTelemetry = new RequestTelemetry();
+            var activity = new Activity(ActivityCreatedByHostingDiagnosticListener);
+
+            StringValues requestId;
+            StringValues standardRootId;
+            if (httpContext.Request.Headers.TryGetValue(RequestResponseHeaders.RequestIdHeader, out requestId))
+            {
+                activity.SetParentId(requestId);
+                requestTelemetry.Context.Operation.ParentId = requestId;
+
+                string[] baggage = httpContext.Request.Headers.GetCommaSeparatedValues(RequestResponseHeaders.CorrelationContextHeader);
+                if (baggage != StringValues.Empty)
+                {
+                    foreach (var item in baggage)
+                    {
+                        NameValueHeaderValue baggageItem;
+                        if (NameValueHeaderValue.TryParse(item, out baggageItem))
+                        {
+                            activity.AddBaggage(baggageItem.Name, baggageItem.Value);
+                            if (!requestTelemetry.Context.Properties.ContainsKey(baggageItem.Name))
+                            {
+                                requestTelemetry.Context.Properties[baggageItem.Name] = baggageItem.Value;
+                            }
+                        }
+                    }
+                }
+            }
+            else if (httpContext.Request.Headers.TryGetValue(RequestResponseHeaders.StandardRootIdHeader, out standardRootId))
```

Comment:
It's not the same. In new events code path, we create another activity. In old events code path, only one Activity is created for both old and new headers.